### DISCUSSION TITLE
Passing Nested Data Parameters to Partials

### DISCUSSION THREAD

**Discussion by ldrummond** [handlebars-lang/handlebars.js#1778] [Q&A]
Hello! 
I have a question that I am bringing here only because I couldn't find documentation or other references to this. 
Is it possible to overwrite a nested data parameter for a partial (` {{> partial user.name='Laura'}}`)? Or is this something that requires a helper?
[Js Fiddle example](https://jsfiddle.net/ldrum/hs2ngjfw/)

Code that fails to compile 
```html
var data = {
  "user": {
    "name": "Bob"
  }
}

Handlebars.registerPartial("user_partial", "{{user.name}}"); 

// Overwrite the "user.name" parameter when rendering the partial
var template = Handlebars.compile(
  "My name is {{> user_partial user[name]='Laura'}}"  // throws compiler error
); 
document.getElementById('output').innerHTML = template(data);
```

Desired output
```html
My name is Laura
```

Thanks

**Comment by jaylinski** [comment]
You can use Partial Contexts for achieving this: https://handlebarsjs.com/guide/partials.html#partial-contexts

```
My name is {{user.name}}

My name is {{> user_partial overrideContext}}
```

```
{
  "user": {
    "name": "Bob"
  },
  "overrideContext": {
    "user": {
      "name": "Bobby"
    }
  }
}
```

### Example

https://handlebarsjs.com/playground.html#format=1&currentExample=%7B%22template%22%3A%22My%20name%20is%20%7B%7Buser.name%7D%7D%5Cn%5CnMy%20name%20is%20%7B%7B%3E%20user_partial%20overrideContext%7D%7D%22%2C%22partials%22%3A%5B%5D%2C%22input%22%3A%22%7B%5Cn%20%20%5C%22user%5C%22%3A%20%7B%5Cn%20%20%20%20%5C%22name%5C%22%3A%20%5C%22Bob%5C%22%5Cn%20%20%7D%2C%5Cn%20%20%5C%22overrideContext%5C%22%3A%20%7B%5Cn%20%20%20%20%5C%22user%5C%22%3A%20%7B%5Cn%20%20%20%20%20%20%5C%22name%5C%22%3A%20%5C%22Bobby%5C%22%5Cn%20%20%20%20%7D%5Cn%20%20%7D%5Cn%7D%22%2C%22output%22%3A%22My%20name%20is%20Bob%5Cn%5CnMy%20name%20is%20Bobby%22%2C%22preparationScript%22%3A%22Handlebars.registerPartial(%5C%22user_partial%5C%22%2C%20%5C%22%7B%7Buser.name%7D%7D%5C%22)%3B%20%22%2C%22handlebarsVersion%22%3A%224.7.7%22%7D

**Reply by usersagent** [reply]
🫡
